src/Microsoft.AspNetCore.Sockets/Transports/LongPollingTransport.cs

```diff
@@ -43,39 +46,28 @@ public LongPollingTransport(ReadableChannel<Message> application, ILoggerFactory
                 context.Response.ContentType = MessageFormatter.GetContentType(messageFormat);
 
                 var writer = context.Response.Body.AsPipelineWriter();
-                var alloc = writer.Alloc(minimumSize: 1);
-                alloc.WriteBigEndian(MessageFormatter.GetFormatIndicator(messageFormat));
+                var output = new PipelineTextOutput(writer, TextEncoder.Utf8); // We don't need the Encoder, but it's harmless to set.
+
+                output.Append(MessageFormatter.GetFormatIndicator(messageFormat));
 
                 while (_application.TryRead(out var message))
                 {
-                    var buffer = alloc.Memory.Span;
-
                     _logger.LogDebug("Writing {0} byte message to response", message.Payload.Length);
 
-                    // Try to format the message
-                    if (!MessageFormatter.TryFormatMessage(message, buffer, messageFormat, out var written))
+                    if (!MessageFormatter.TryWriteMessage(message, output, messageFormat))
                     {
-                        // We need to expand the buffer
-                        // REVIEW: I'm not sure I fully understand the "right" pattern here...
-                        alloc.Ensure(MaxBufferSize);
-                        buffer = alloc.Memory.Span;
+                        // We ran out of space to write, even after trying to enlarge.
+                        // This should only happen in a significant lack-of-memory scenario.
 
-                        // Try one more time
-                        if (!MessageFormatter.TryFormatMessage(message, buffer, messageFormat, out written))
-                        {
-                            // Message too large
-                            throw new InvalidOperationException($"Message is too large to write. Maximum allowed message size is: {MaxBufferSize}");
-                        }
+                        // IOutput doesn't really have a way to write incremental
+
+                        // REVIEW: I really am not sure what exception to throw here... OOM seems right, but it feels so wrong to throw it ourselves...
+                        throw new OutOfMemoryException("Ran out of space to format messages!");
```

Comment:
Yeah... it just seems like `InvalidOperationException` is no better than just going `throw new Exception(...)` here. I'll swap it.